refactor(napi/oxlint): remove classes

overlookmotel opened the pull request:
Flatten code in `napi/oxlint2`, removing the `PluginRegistry` and `Linter` classes.

Only a single instance of these classes was created, so the classes added nothing in terms of utility, and added a (small) perf cost.

The classes *did* provide a nice encapsulation of concerns, but in my opinion our aim with the JS code here should be performance above all else. I hope we can avoid the code descending into a spaghetti monstrosity by breaking things up into separate files and with good documentation (I assume we'll bundle and minify this code at some point, so splitting code into multiple files and verbose comments won't have a cost at run time).

camc314 commented:
### Merge activity

* **Jul 22, 9:34 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(napi/oxlint): remove classes (#12454)

Flatten code in `napi/oxlint2`, removing the `PluginRegistry` and `Linter` classes.

Only a single instance of these classes was created, so the classes added nothing in terms of utility, and added a (small) perf cost.

The classes *did* provide a nice encapsulation of concerns, but in my opinion our aim with the JS code here should be performance above all else. I hope we can avoid the code descending into a spaghetti monstrosity by breaking things up into separate files and with good documentation (I assume we'll bundle and minify this code at some point, so splitting code into multiple files and verbose comments won't have a cost at run time).
